service/controller/v17/resource/cloudformation/delete.go

```diff
@@ -51,7 +51,7 @@ func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange inte
 		}
 
 		if r.encrypterBackend == encrypter.VaultBackend {
-			err = r.removeRoleAccess(sc, customObject)
+			err = r.encrypterRoleManager.EnsureDeletedAuthorizedIAMRoles(ctx, customObject)
```

Comment:
Now this is the only dependency to the vault encrypter in `cloudformation` package.